Fix the 2.4.x branch build when DNS-SD is disabled.

Without HAVE_DNSSD, data is a _cups_getdata_t and does not contain the
num_local or local_dests members used only to suppress duplicate DNS-SD
discoveries.

--- a/cups/dest.c
+++ b/cups/dest.c
@@ -3574,8 +3574,10 @@ cups_enum_dests(
     */
 
     num_dests = _cupsGetDests(http, IPP_OP_CUPS_GET_PRINTERS, NULL, &dests, type, mask);
+#ifdef HAVE_DNSSD
     data.num_local   = 0;
     data.local_dests = NULL;
+#endif /* HAVE_DNSSD */
 
     if (data.def_name[0])
     {
@@ -3603,7 +3605,9 @@ cups_enum_dests(
       * Pass the list of local printers to the callback function...
       */
 
+#ifdef HAVE_DNSSD
       data.num_local = cupsCopyDest(dest, data.num_local, &data.local_dests);
+#endif /* HAVE_DNSSD */
 
       if ((user_dest = cupsGetDest(dest->name, NULL, data.num_dests, data.dests)) != NULL)
       {
@@ -3978,7 +3982,9 @@ cups_enum_dests(
   enum_finished:
 
   cupsFreeDests(data.num_dests, data.dests);
+#ifdef HAVE_DNSSD
   cupsFreeDests(data.num_local, data.local_dests);
+#endif /* HAVE_DNSSD */
 
 #ifdef HAVE_DNSSD
   cupsArrayDelete(data.devices);
